File: Src/Icm.Core.Tests/Collections extensions/ICollectionExtensionsTest.vb
Imports Icm.Collections

<Category("Icm")>
<TestFixture()>
Public Class ICollectionExtensionsTest

    <TestCase({"a", "b", "c"}, ";", "a;b;c")>
    <TestCase({"a", "b", "c"}, "", "abc")>
    <TestCase({"a", "b", "c"}, Nothing, "abc")>
    <TestCase({"a", Nothing, "c"}, ";", "a;;c")>
    <TestCase({"a", "", "c"}, ";", "a;;c")>
    <TestCase({"asdf"}, ";", "asdf")>
    <TestCase({}, ";", "")>
    Public Sub JoinStrTest(col As IEnumerable(Of String), separator As String, expected As String)
        Dim actual As String

        actual = col.JoinStr(separator)
        Assert.That(expected, [Is].EqualTo(actual))
    End Sub

    <Test(), Category("Icm")>
    Public Sub ForceRemoveTest()

        Dim c As ICollection(Of String) = Nothing
        Dim item As String = "b"

        c = New List(Of String) From {"a", "b", "c"}
        ICollectionExtensions.ForceRemove(Of String)(c, item)
        Assert.IsFalse(c.Contains(item))

        'Caso 1
        c = New List(Of String) From {"a", "b", "c"}
        item = "x"
        ICollectionExtensions.ForceRemove(Of String)(c, item)
        Assert.IsFalse(c.Contains(item))

        'Caso 2
        c = New List(Of String) From {"a", "b", "c"}
        item = Nothing
        ICollectionExtensions.ForceRemove(Of String)(c, item)
        Assert.IsFalse(c.Contains(item))

        'Caso 3
        c = New List(Of String) From {"a", "b", "c"}
        item = ""
        ICollectionExtensions.ForceRemove(Of String)(c, item)
        Assert.IsFalse(c.Contains(item))

        'Caso 4
        c = New List(Of String) From {""}
        item = "b"
        ICollectionExtensions.ForceRemove(Of String)(c, item)
        Assert.IsFalse(c.Contains(item))

    End Sub
End Class

File: Src/Icm.Core.Tests/Collections extensions/ICollectionExtensionsTest.cs

using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using Icm.Collections;
using NUnit.Framework;

[Category("Icm")]
[TestFixture()]
public class ICollectionExtensionsTest
{

	[TestCase({
		"a",
		"b",
		"c"
	}, "b", {
		"a",
		"c"
	})]
	[TestCase({
		"a",
		"b",
		"c"
	}, "x", {
		"a",
		"b",
		"c"
	})]
	[TestCase({
		"a",
		null,
		"c"
	}, null, {
		"a",
		"c"
	})]
	[TestCase({
		"a",
		"",
		"c"
	}, "", {
		"a",
		"c"
	})]
	[TestCase(new string[], "a", new string[])]
	[TestCase(new string[], null, new string[])]
	[TestCase(new string[], "", new string[])]
	public void ForceRemove_NormalTests(IEnumerable<string> col, string itemRemoved, IEnumerable<string> expected)
	{
		List<string> list = new List<string>(col);
		list.ForceRemove(itemRemoved);
		Assert.That(list, Is.EquivalentTo(expected));
	}

	[Test()]
	public void ForceRemove_WithNullArray_ThrowsNullReferenceException()
	{
		ICollection<string> col = null;

		Assert.That(() => col.ForceRemove(";"), Throws.TypeOf<NullReferenceException>);
	}

}

//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================
